Fenicsx v0.5.0

ma595 opened the pull request:
Update fenicsx packages to latest tagged releases (v0.5.0).

ma595 committed:
initial commit of 0.5.0 changes

ma595 committed:
updated dependencies

ma595 committed:
updated ffcx sha

ma595 committed:
comment style

ma595 committed:
llvm compilers

ma595 committed:
introduce pugixml dependency for 0.5.0:

chrisrichardson committed:
Merge branch 'spack:develop' into fenicsx_v0.5.0

ma595 committed:
update compilers to support C++20 features

ma595 committed:
Merge branch 'fenicsx_v0.5.0' of github.com:FEniCS/spack into fenicsx_v0.5.0

garth-wells committed:
Merge branch 'spack:develop' into fenicsx_v0.5.0

ma595 committed:
style fixes

ma595 committed:
Merge branch 'fenicsx_v0.5.0' of github.com:FEniCS/spack into fenicsx_v0.5.0

garth-wells commented on var/spack/repos/builtin/packages/fenics-basix/package.py:
```diff
@@ -15,11 +15,21 @@ class FenicsBasix(CMakePackage):
+    )
+
     depends_on("cmake@3.18:", type="build")
     depends_on("xtl@0.7.2:")
```
Basix v0.5 no longer depends on `xtl` or `xtensor`.

garth-wells commented on var/spack/repos/builtin/packages/py-fenics-basix/package.py:
```diff
@@ -16,12 +16,14 @@ class PyFenicsBasix(PythonPackage):
     version("0.1.0", sha256="2ab41fe6ad4f6c42f01b17a6e7c39debb4e0ae61c334d1caebee78b741bca4e7")
 
     depends_on("fenics-basix@main", type=("build", "run"), when="@main")
+    depends_on("fenics-basix@0.5.0", type=("build", "run"), when="@0.5.0")
```
v0.5 no longer depends on `xtl` or `xtensor`.

garth-wells reviewed (changes requested):
Looks good, I've left some inline comments.

`(py)-fenics-dolfinx@main` no longer depends on `xtl` and `xtensor`.

Has this PR been tested using the CI at https://github.com/FEniCS/dolfinx/actions/workflows/spack.yml?

ma595 committed:
xtensor and xtl not needed for basix 0.5.0 and above

garth-wells committed:
Merge branch 'develop' into fenicsx_v0.5.0

garth-wells committed:
Skip to Basix 0.5.1

The 0.5.1 release removes the C++ build dependency on Python that sneaked into the 0.5.0 build system.

garth-wells committed:
Improve depends on version ranges

garth-wells committed:
More dependency version improvements

garth-wells commented:
@tldahlgren I think this is ready to merge now.

jhale commented:
I have a slightly refined version of this patch that better reflects some upcoming changes on `main` and marks some of the older versions as deprecated which will allow us to reduce complexity of these files in the future. I will push it to a new PR shortly.

tldahlgren reviewed (approved):
Confirmed all new sha256.
